php/libraries/SinglePointLogin.class.inc

```diff
@@ -316,6 +317,16 @@ class SinglePointLogin
             $this->_lastError = "Incorrect username or password.";
             return false;
         }
+        /* Check whether a user's account is locked due to too many bad login
+         * attempts before actually trying to authenticate their credentials.
+         * Doing so can help mititgate timing attacks against the server.
```

Comment:
The last sentence is wrong. To mitigate timing attacks you need to ensure that all code paths are constant time and there's no branching/early return.